export/wiki/stage_items.py

```diff
@@ -1,113 +1,165 @@
-from typing import Any, Dict, Optional, cast
+from typing import Any, Dict, List, Optional, cast
 
 from ark.types import PrimalItem
-from automate.hierarchy_exporter import JsonHierarchyExportStage
+from automate.hierarchy_exporter import ExportFileModel, ExportModel, Field, JsonHierarchyExportStage
 from ue.asset import UAsset
+from ue.hierarchy import get_parent_class
+from ue.properties import FloatProperty, IntProperty, ObjectProperty
 from ue.proxy import UEProxyStructure
 from utils.log import get_logger
 
-from .items.cooking import convert_cooking_values
-from .items.crafting import convert_crafting_values
+from .flags import gather_flags
+from .items.cooking import CookingIngredientData, convert_cooking_values
+from .items.crafting import CraftingData, RepairData, convert_crafting_values
 from .items.durability import convert_durability_values
-from .items.egg import convert_egg_values
-from .items.status import convert_status_effect
+from .items.egg import EggData, convert_egg_values
+from .items.status import StatEffectData, convert_status_effects
 
 __all__ = [
     'ItemsStage',
 ]
 
 logger = get_logger(__name__)
 
+OUTPUT_FLAGS = (
+    'bPreventCheatGive',
+    'bDurabilityRequirementIgnoredInWater',
+)
+
+
+class Item(ExportModel):
+    name: Optional[str]
+    description: Optional[str] = None
+    bp: str = Field(..., title="Blueprint path")
+    parent: Optional[str] = Field(None, description="Full path to the parent class of this item")
+    icon: Optional[str] = Field(
+        None,
+        title="Icon blueprint path",
+        description="This is either a texture or a material instance",
+    )
+    type: Optional[str] = None
+    flags: Optional[List[str]] = Field(
+        list(),
+        description="Relevant boolean flags that are True for this item",
+    )
+    folders: List[str] = Field(
+        [],
+        title="Crafting station folder",
+        description="These are the folders in a crafting station where this item's blueprint is shown.",
+    )
+    weight: Optional[FloatProperty] = Field(None, title="Weight of a single unit")
+    stackSize: Optional[IntProperty] = None
+    spoilsIn: Optional[FloatProperty] = Field(None, title="Spoilage time")
+    spoilsTo: Optional[str] = Field(None, title="Spoilage product")
+    durability: Optional[float] = Field(None, title="Durability units", description="At 100% quality")
+    crafting: Optional[CraftingData]
+    repair: Optional[RepairData]
+    structure: Optional[str] = Field(
+        None,
+        title="Structure blueprint path",
```

Comment:
This title is unnecessary.
Description: "Blueprint path of the structure created by this item (can by looked up in structures.json)."